docs/getting-started/interactivity.mdx

```diff
@@ -140,48 +140,78 @@ In the above example, whenever the `Input` field changes, it updates the value o
 
-#### `UpdateVariable`
+The primary way to create actions in Dara is the `@action` decorator. This decorator takes a function and returns an action that can be passed to a component's callback.
+It injects an `ActionCtx` object as the first argument of the function, which contains the input sent from the component and exposes action methods to update the state of the application.
 
-The `dara.core.interactivity.actions.UpdateVariable` action can be passed to any component prop accepting an Action and trigger the update of a `dara.core.interactivity.plain_variable.Variable`, `dara.core.interactivity.data_variable.DataVariable` or `dara.core.interactivity.url_variable.UrlVariable`.
-
-It takes the following arguments:
+:::note
 
-- `resolver`: a function to resolve the new value for the `Variable`, `DataVariable` or `UrlVariable`. The resolver takes one argument: a context of type `UpdateVariable.Ctx`. The new value is given by the component and can be obtained with `ctx.inputs.new`. While the current value, or now previous value of the variable you are updating, can be obtained with `ctx.inputs.old`.
-- `variable`: the `Variable`, `DataVariable` or `UrlVariable` to update with a new value upon triggering the action
-- `extras`: any extra variables to resolve and pass to the resolver function, you can obtain a list of the resolved values of all the extras passed with `ctx.extras` in the resolver function
+The action methods on `ActionCtx` are `async` and must be `await`-ed, so your decorated function should be `async` as well.
 
-To see how the `UpdateVariable` works, consider the example below:
+:::
 
 ```python
-from dara.core import Variable, UpdateVariable
-from dara.components import Button, Stack
+from dara.core import action, Variable
+from dara.components import Select, Item
 
-my_var = Variable(5)
+some_variable = Variable()
 
-def add_one(ctx: UpdateVariable.Ctx):
-    return ctx.inputs.old + 1
+@action
+async def my_action(ctx: action.Ctx, inp):
+    # Value coming from the component, in this case the selected item
+    value = ctx.input
+    # Your action logic...
+
+    # Update `some_variable` to `value` multiplied by 2
+    await ctx.update(target=some_variable, value=value * 2)
 
-def test_page():
-    return Stack(
-        Text(my_var),
-        Button(
-            'Add One',
-            onclick=UpdateVariable(add_one, variable=my_var)
-        ),
-    )
 
-config.add_page(name='Increment Variable', content=test_page())
+Select(
+    items=[Item(label='item1', value=1), Item(label='item2', value=2)],
+    onchange=my_action(2)
+)
 ```
 
-The `dara.components.common.button.Button` component has a parameter `onclick` which is the component's callback. You can pass it an action, which in this case is an `UpdateVariable` instance.
+The example above shows how to use the `@action` decorator to create an action that updates a `Variable` with the value of the selected item in a `Select` component.
```

Comment:
I'm a bit confused here. I am guessing `value=value * 2` was supposed to be `value=value * inp` and if this is the case I would explain better what this example is doing which is rather updating a variable with the value of the selected item multiplied by 2

Reply:
yep looks good!